patcher/Patcher.py

```
@@ -324,84 +238,81 @@ def align(v, a):
+        config.gcc_path, "-c", *config.asm_flags, build_hooks_folder / "*.cpp"
+    ), cwd=build_hooks_folder):
+        raise Exception(
+            "Errors occurred during building of generated hooks files")
+
+    for hook_path in list_files_at(hooks_folder_path, "**/*.cpp"):
+        print(f"Hook file {hook_path} is deprecated, use .hook file instead")
+
+    if run_process(
+        (config.gcc_path, "-c", *config.asm_flags, hooks_folder_path / "*.cpp"),
+            cwd=build_hooks_folder):
         raise Exception("Errors occurred during building of hooks files")
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)